query.py

```diff
@@ -97,143 +128,157 @@ def create_hubmap_metadata_df(
             "[cyan]Processing datasets...", total=len(hubmap_metadata), uuid=""
         )
-        for uuid in hubmap_metadata["uuid"].values:
+
+        for uuid in metadata_lookup.index:
             progress.update(task, uuid=uuid)
-            dataset_info = get_dataset_info(uuid)[0]
-            donor_metadata = (
-                dataset_info.get("donor", {})
-                .get("metadata", {})
-                .get("organ_donor_data", [])
-            )
-
-            dataset_urls = get_dataset_urls(
-                uuid, file_types=file_types, dataset_class=dataset_class
-            )
-            urls = {}
-            for attr in dir(dataset_urls):
-                if not attr.startswith("_") and not callable(
-                    getattr(dataset_urls, attr)
-                ):
-                    value = getattr(dataset_urls, attr)
-                    urls[f"{attr}_url"] = value if value is not None else ""
-
-            row = {
-                "assay": hubmap_metadata.loc[
-                    hubmap_metadata["uuid"] == uuid, "assay_type"
-                ].iloc[0],
-                "rnaseq_assay_method": hubmap_metadata.loc[
-                    hubmap_metadata["uuid"] == uuid, "rnaseq_assay_method"
-                ].iloc[0],
-                "title": dataset_info.get("title", ""),
-                "group_name": dataset_info.get("group_name", ""),
-                "consortium": "HuBMAP",
-                "doi": dataset_info.get("registered_doi", ""),
-                "publication_date": datetime.fromtimestamp(
-                    dataset_info.get("published_timestamp", 0) / 1000, tz=timezone.utc
-                ).strftime("%Y-%m-%d")
-                if dataset_info.get("published_timestamp")
-                else "",
-                "status": dataset_info.get("data_access_level", ""),
-                "dataset_type": dataset_info.get("dataset_type", ""),
-                "processing": "raw",
-                "organ": next(
-                    (
-                        sample.get("organ", "")
-                        for sample in dataset_info.get("origin_samples", [])
-                        if sample.get("organ")
+
+            try:
+                dataset_info = get_dataset_info(uuid)[0]
+                donor_metadata = (
+                    dataset_info.get("donor", {})
+                    .get("metadata", {})
+                    .get("organ_donor_data", [])
+                )
+
+                dataset_urls = get_dataset_urls(
+                    uuid,
+                    file_types=file_types,
+                    dataset_class=dataset_class,
+                )
+
+                if dataset_urls is None:
+                    logger.warning(
+                        f"No usable files for uuid {uuid}, HuBMAP ID {dataset_info.get('hubmap_id')}."
+                    )
+                    continue
+
+                # Dynamically collect all URLs from dataset class
+                urls = {
```

Comment:
It's a bit weird that we're basically getting the URLs twice here. Here and just above.